### PR TITLE
Implement user registration and login functionality with password encryption

### PR DESCRIPTION
**Pull request by mayaajike**

### CODE REVIEW

**Comment by mellyeliu** [comment]
Great start and progress within such a short timestamp! 
- Really appreciate the well structured DB, as well as the careful use of primary and foreign keys
- Can we update the PR with screenshots of the current behaviour? Also in the future, could we separate out large changes like installing packages or generating base files from the PR? it can cause extra noise

Some small feedback below:

File: Frontend/src/Components/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                },
+                body: JSON.stringify({ firstName, lastName, username, email, password }),
+                credentials: 'include'
+            });
+
+            if (response.ok) {
+                const data = await response.json();
+                const loggedInUser = data.user;
+
+                console.log('Signup Successful!')
```
Comment (mellyeliu) [inline Frontend/src/Components/Signup.jsx]: let's remove any logs from reaching prod

File: Frontend/src/Components/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        } catch (error) {
+            alert('Signup failed' + error);
+        }
+    }
+
+    return (
+        <div className='signup-container'>
+            <form className='signup-form' onSubmit={handleSubmit}>
+                <h2>Sign Up</h2>
+                <div className='form-group'>
```
Comment (mellyeliu) [inline Frontend/src/Components/Signup.jsx:54]: can we add these values to a map then iterate through them? that'll keep things a bit cleaner and allow you to add more form values with ease

File: Frontend/src/Components/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                const data = await response.json();
+                const loggedInUser = data.user;
+
+                console.log('Signup Successful!')
+
+                setFirstName('');
+                setLastName('');
+                setUsername('');
+                setEmail('');
+                setPassword('');
```
Comment (mellyeliu) [inline Frontend/src/Components/Signup.jsx:37]: do we have any password validation? min length, numbers, etc?

File: Backend/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  id                    String            @id @default(uuid())
+  firstName             String
+  lastName              String
+  username              String            @unique
+  password              String
+  email                 String            @unique
+  spotifyAccount        SpotifyAccount?
+  nowListening          NowListening[]
+  initiatedFriendships  Friendship[]      @relation("FriendshipInitiator")
+  receivedFriendships   Friendship[]      @relation("FriendshipReceiver")
```
Comment (mellyeliu) [inline Backend/prisma/schema.prisma:27]: are these friend requests received, but not yet accepted?

File: Backend/prisma/schema.prisma [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+
+model User {
```
Comment (mellyeliu) [inline Backend/prisma/schema.prisma:17]: is there a reason there's no list of confirmed friends in this model?

**Review by mellyeliu — CHANGES_REQUESTED** [review]

File: Backend/prisma/schema.prisma (hunk 4 above)
Comment (mayaajike) [reply Backend/prisma/schema.prisma:27]: yes

**Comment by mayaajike** [comment]
Screenshot 2024-06-28 at 6 34 49 AM
Screenshot 2024-06-28 at 6 35 31 AM
Screenshot 2024-06-28 at 7 06 49 AM

**Comment by mellyeliu** [comment]
LGTM! Thanks for making the changes to the password validation

**Review by mellyeliu — APPROVED** [review]

**Pull request merged**